huggingface_pipelines/text.py

```diff
@@ -0,0 +1,161 @@
+from sonar.inference_pipelines.text import EmbeddingToTextModelPipeline
+import logging
+from typing import List, Dict, Any, Union
+from .pipeline import Pipeline, PipelineOverwrites, PipelineConfig
+from sonar.inference_pipelines.text import TextToEmbeddingModelPipeline
+import torch
+from dataclasses import dataclass, replace
+import numpy as np
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class TextToEmbeddingOverwrites(PipelineOverwrites, total=False):
+    encoder_model: str
+    source_lang: str
+
+
+class EmbeddingToTextOverwrites(PipelineOverwrites, total=False):
+    decoder_model: str
+    target_lang: str
+
+
+@dataclass
+class TextToEmbeddingPipelineConfig(PipelineConfig):
+    """
+    Configuration class for text-to-embedding pipelines.
+
+    Attributes:
+        encoder_model (str): The name or path of the model to be used for encoding texts into embeddings.
+        source_lang (str): The source language code for the texts to be encoded.
+    """
+    encoder_model: str = "text_sonar_basic_encoder"
+    source_lang: str = "eng_Latn"
+
+    def with_overwrites(self, overwrites: TextToEmbeddingOverwrites):
+        return replace(self, **overwrites)
```

Comment:
is it really specific to  TextToEmbeddingPipelineConfig? why not keep it at PipelineConfig level